Jenkinsfile

```diff
@@ -1,27 +1,392 @@
+        string(defaultValue: "c2d53661-8521-47c7-a7b3-73bbb6723c0a", name: "WIN_CERT_PASS_CREDENTIAL_ID"),
+        string(defaultValue: "44c2aee7-743e-4ede-9411-55ad7219b09c", name: "WIN_CERT_PATH_CREDENTIAL_ID"),
+        string(defaultValue: "761dc30d-f04f-49a5-9940-cdd8ca305165", name: "MAC_CERT_CREDENTIAL_ID"),
+        string(defaultValue: "3428e3e4-5733-4e59-8c6b-f95f1ee00322", name: "MAC_CERT_PASS_CREDENTIAL_ID"),
+        string(defaultValue: "761dc30d-f04f-49a5-9940-cdd8ca305165", name: "MAR_CERT_CREDENTIAL_ID"),
+        string(defaultValue: "3428e3e4-5733-4e59-8c6b-f95f1ee00322", name: "MAR_CERT_PASS_CREDENTIAL_ID"),
+        string(defaultValue: "debian-gpg-key", name: "DEBIAN_GPG_KEY_CREDENTIAL_ID"), 
+        string(defaultValue: "debian-gpg-pass", name: "DEBIAN_GPG_PASS_CREDENTIAL_ID"),
+        string(defaultValue: "1.11.0", name: "CQZ_VERSION"),
+        booleanParam(defaultValue: false, description: '', name: 'MAC_REBUILD_IMAGE'),
+        booleanParam(defaultValue: false, description: '', name: 'WIN_REBUILD_IMAGE'),
+        booleanParam(defaultValue: false, description: '', name: 'LIN_REBUILD_IMAGE'),
+    ]), 
+    pipelineTriggers([])
+])
 
-node('browser') {
-  ws('build') {
-    stage('checkout') {
-      checkout scm
+node('docker') {    
+    stage("Copy XPI") {
+        UPLOAD_PATH="s3://repository.cliqz.com/dist/$CQZ_RELEASE_CHANNEL/${params.CQZ_VERSION}/${CQZ_BUILD_ID}/cliqz@cliqz.com.xpi"
+        HTTPSE_UPLOAD_PATH="s3://repository.cliqz.com/dist/$CQZ_RELEASE_CHANNEL/${params.CQZ_VERSION}/${CQZ_BUILD_ID}/https-everywhere@cliqz.com.xpi"
+
+        withCredentials([
+            [$class: 'AmazonWebServicesCredentialsBinding',
+            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+            credentialsId: params.CQZ_AWS_CREDENTIAL_ID,
+            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+
+            sh "aws s3 cp ${params.CQZ_EXTENSION_URL} $UPLOAD_PATH"
+            sh "aws s3 cp ${params.CQZ_HTTPSE_EXTENSION_URL} $HTTPSE_UPLOAD_PATH"
+        }
+    }       
+}
+
+jobs["windows"] = {
+    node('docker') {
+        ws() {
+            stage('GPU Slave Docker Checkout') {
+                checkout scm
+            }
+            try {
+               helpers = load "build-helpers.groovy"
+            } catch(e) {
+                echo "Could not load build-helpers"
+                throw e
+            }    
+        }
     }
+    // Try to get a jenkins slave. If the slave was created by this action
+    // we need to bootstrap it with ansible
+    def ec2_node = helpers.getEC2Slave("c:/jenkins")
+
+    if (ec2_node.get('created')) {
+        echo "New slave created. Starting provisioning"
+
+        node('docker') {
+            ws() {
+                withCredentials([
+                [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: params.CQZ_AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
```

Comment:
please wrap the long lines. 

we can discuss a code style for groovy if you like